src/librustc/mir/mod.rs

```diff
@@ -212,26 +210,54 @@ impl<'tcx> Body<'tcx> {
 
     #[inline]
     pub fn basic_blocks_mut(&mut self) -> &mut IndexVec<BasicBlock, BasicBlockData<'tcx>> {
-        self.cache.invalidate();
+        debug!("Clearing predecessors cache at: {:?}", self.span.data());
+        self.predecessors_cache = None;
         &mut self.basic_blocks
     }
 
     #[inline]
     pub fn basic_blocks_and_local_decls_mut(
         &mut self,
     ) -> (&mut IndexVec<BasicBlock, BasicBlockData<'tcx>>, &mut LocalDecls<'tcx>) {
-        self.cache.invalidate();
```

Comment:
This also needs to invalidate

Reply:
Woops, I missed that one when cleaning up. Thanks for catching that! (I now might have more panics to address)